### PR TITLE
fix: executor plugin sidecar can be injected with sa secrets.

### PR DESCRIPTION
**Pull request by berlinsaint**
Signed-off-by: maybaby <berlinsaint@126.com>

Fixes #TODO

### CODE REVIEW

**Comment by berlinsaint** [comment]
By my testing, i found that it create sidecar with `automountServiceAccountToken` set to false, this means no easy way to communitate with k8s api if want do sth in plugin.

File: workflow/controller/agent.go [hunk 1]
```diff
@@ -135,7 +135,7 @@ func (woc *wfOperationCtx) createAgentPod(ctx context.Context) (*apiv1.Pod, erro
 				RunAsUser:    pointer.Int64Ptr(8737),
 			},
 			ServiceAccountName:           serviceAccountName,
-			AutomountServiceAccountToken: pointer.BoolPtr(false),
+			AutomountServiceAccountToken: pointer.BoolPtr(*woc.execWf.Spec.AutomountServiceAccountToken),
```
Comment (simster7) [inline workflow/controller/agent.go:138]: Why de-reference and re-reference? Simpler:

```suggestion
			AutomountServiceAccountToken: woc.execWf.Spec.AutomountServiceAccountToken,
```

**Review by simster7 — COMMENTED** [review]
Minor change otherwise LGTM

File: workflow/controller/agent.go (hunk 1 above)
Comment (alexec) [reply workflow/controller/agent.go:138]: I think this could change the default behaviour to mount the token. We don’t want that.

Comment (berlinsaint) [reply workflow/controller/agent.go:138]: do you mean that the sidecar(plugin)is designed to that cant mount sa by default?  why ？

Comment (alexec) [reply workflow/controller/agent.go:138]: Correct. Security.

Comment (berlinsaint) [reply workflow/controller/agent.go:138]: one more questuon,how can i init k8s api in sidecar?  Or as I mentioned in [here](https://github.com/argoproj/argo-workflows/issues/7860)
is there a design to make the plugin run forever such as using deployment?

**Comment by alexec** [comment]
Ok, so this shows you how to do secrets:

https://github.com/argoproj-labs/argo-workflows-slack-executor-plugin

Do we need to add docs?

**Comment by berlinsaint** [comment]
> Ok, so this shows you how to do secrets:
> 
> https://github.com/argoproj-labs/argo-workflows-slack-executor-plugin
> 
> Do we need to add docs?

I think what i need is  plugin authentication as [8144](https://github.com/argoproj/argo-workflows/issues/8144) says?

**Pull request closed without merging**